src/go/k8s/api/redpanda/v1alpha1/redpanda_conversion_test.go

```diff
@@ -151,3 +217,32 @@ func createOldRedpanda(flag *v1alpha1.CloudStorageEnabledString) *v1alpha1.Redpa
 	}
 	return r
 }
+
+func MustMarshal(t *testing.T, obj any) []byte {
+	out, err := json.Marshal(obj)
+	require.NoError(t, err)
+	return out
+}
+
+func CRDFuzzer() *fuzz.Fuzzer {
```

Comment:
I missed that resource Quantity is problematic.

Could this fuzzer be used in `newV1alpha2` and `newV1alpha1` function? 

With that weird loop, which is using fuzzer, could be removed.